matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java

```diff
@@ -513,15 +528,22 @@ private JSONObject getOwnership()
                 userIsOwner = true;
             } else {
                 Set<Group> userGroups = GROUP_MANAGER.getGroupsForUser(currentUser);
+                XWikiContext context = CONTEXT_PROVIDER.get();
                 for (Group group : userGroups) {
                     DocumentReference groupRef = group.getReference();
                     if (groupRef.equals(ownerRef)) {
                         userGroupIsOwner = true;
+                        break;
+                    }
+
+                    if (userIsGroupManager(userRef, groupRef, context)) {
```

Comment:
this does not find if current user is the administrator of the match owner's group, it just finds that current user is administrator of some group

Reply:
...btw, I think to speed things up for admins (who get to see 1000s of matches) we can say that if current user is admin `userIsGroupManager` is always true regardless of actual match owner. That way for admins we will never run the expensive "is group admin" check, and for users who only see a few matches it is not  anbig deal to run it